AK/Platform.h

```diff
@@ -27,6 +28,18 @@
 #    define AK_OS_BSD_GENERIC
 #endif
 
+#if defined(__CYGWIN__) || defined(_WIN32)
+#    define AK_OS_WIN32
+// FIXME: Do the following properly
+#    define O_CLOEXEC FD_CLOEXEC
+#    undef __POSIX_VISIBLE
+#    define __POSIX_VISIBLE 200809
```

Comment:
We should not be setting any of these manually. The cannonical way of doing this would be to 

```cmake
if (WIN32)
  add_compile_definitions(_POSIX_SOURCE=200809)
  add_compile_definitions(_DEFAULT_SOURCE)
endif()
```

Reply:
I can't get that to work. Probably gets overwritten by cygwin's `features.h` so this hacky way is the way to go till somebody feels like figuring out how to do it properly cygwin-style.